### ISSUE TITLE
Set manually the destination image size of LinearPolar / LogPolar

### ISSUE THREAD

**Issue by formicant** [opencv/opencv#9346]
##### Detailed description

The methods linearPolar and logPolar return the image of the same size as the source.

I consider this behavior strange since the dimensions of the result image are ρ and φ and have nothing to do with the original image width and height.

Ideally, the radial and angular dimensions should be set manually with some reasonable defaults depending on the _maxRadius_ parameter (say, _maxRadius_ × _π∙maxRadius_ to match the average resolution of the source for LinearPolar). I don’t think, however, that changing the default behavior (and thus breaking the compatibility) is the right thing to do.

So, is it possible to just add the method overloads with the destination image size as an argument to have the ability to set it manually?

**Comment by PkLab** [comment]
I agree. Is much more natural to have a mapping where cols=ρ and rows=φ  (or viceversa).

While I was working on doc for `linearPolar` I've developed a version that's works are requested by @formicant 

The function uses Angle and Distance amplifier so that output image size is 
- rows = 360 * angle amplifier 
- cols = radius * distance amplifier 

Check it here https://gist.github.com/PkLab/49a6e882806a62ed4315566733e4748b

@sovrasov If you like I can put the code in a pull request

**Comment by PkLab** [comment]
I've updated the Gist. New version of the function allow angles in cols or in rows. 
About resolution, the best would be 1=1 mapping. This means that src circle and dst image would have same area. The new version allows automatic calculation of angle resolution to get 1=1 mapping.

**Comment by sovrasov** [comment]
I think these changes are useful and in future we will remove old variants of the polar functions.

**Comment by Felix-neko** [comment]
Please add this change, it's really useful (for example, I can have a polar image 360x1000 and want to make [2000x2000] cartesian from it).

**Comment by PkLab** [comment]
Hi guys,
please check my pull request  ... I'm waiting there for your comment :)
